pkg/loop/internal/chain_reader.go

```diff
@@ -0,0 +1,286 @@
+package internal
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	"github.com/fxamacker/cbor/v2"
+	libocr "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+	"google.golang.org/grpc/status"
+
+	"github.com/smartcontractkit/chainlink-relay/pkg/loop/internal/pb"
+	"github.com/smartcontractkit/chainlink-relay/pkg/types"
+)
+
+var _ types.ChainReader = (*chainReaderClient)(nil)
+
+type chainReaderClient struct {
+	*brokerExt
+	grpc pb.ChainReaderClient
+}
+
+// enum of all known encoding formats for versioned data
+const (
+	SimpleJsonEncodingVersion = iota
+	CborEncodingVersion
```

Comment:
Let's follow go naming here, and why `Simple`?
```suggestion
	JSONEncodingVersion = iota
	CBOREncodingVersion
```

Reply:
I figured `Simple` might be a good prefix to indicate it's the original version where we just blindly pass it to `json.Marshal` and `json.Unmarshal`... in case at some point in the future we had a more sophisticated version of json encoding where we did some other processing on top of that.  But I guess that's unlikely since we've already moved to CBOR anyway.